src/lib.rs

```diff
@@ -124,16 +146,20 @@ impl AsciiChars {
             let res: u32;
 
             unsafe {
-                asm!("pcmpestri $$0, ($1, $5), $2"
+                asm!("# Move low word of $2 to high word of $1
+                      movlhps $2, $1
+                      pcmpestri $$0, ($3, $4), $1"
                      : // output operands
                      "={ecx}"(res)
                      : // input operands
-                     "r"(ptr),
                      "x"(self.needle),
-                     "{rdx}"(len),
-                     "{rax}"(self.count as u64),
+                     "x"(self.needle_hi),
+                     "r"(ptr),
                      "r"(offset)
+                     "{rdx}"(len),              // haystack length
+                     "{rax}"(self.count as u64) // needle length
                      : // clobbers
+                     "cc"
```

Comment:
You know, I swear I put these clobbers in at some point. Wonder what happened to them...